tools/mkclass/classcompiler.cpp

```diff
@@ -517,15 +517,25 @@ void ClassCompiler::HandleClass(const Klass& klass, const ClassDebugInfo&)
 		}
 
 		if (field.Type.IsName) {
-			m_Impl << "\t" << "if (";
+			if (field.Type.ArrayRank > 0) {
+				m_Impl << "\t\t\t" << "if (!" << valName << ".IsEmpty() && ";
```

Comment:
That part of the condition structure differs from the already existing one. Why?

Reply:
As far as I could figure out, `valName` is already a string type if `field.Type.ArrayRank` isn't > 0. And when `valName` is already a string type, I don't need to test it for.

Reply:
OK, forget it.